File: CSHARP/CSHARP/Model/Partija.cs
using System;
using System.Collections.Generic;

namespace CSHARP.Model
{
    public abstract class Partija: Entitet
    {


        public int doKolikoSeIgra { get; set; }
        public Lokacija lokacija { get; set; }
        public Igrac unosi { get; set; }
        public List<Mjesanje> mjesanja { get; set; }
        public List<Igrac> igraci { get; set; }

        public virtual Rezultat getRezultat()
        {
            var r = new Rezultat();
            r.prvi = 0;
            r.drugi = 0;
            foreach(Mjesanje m in mjesanja)
            {
                r.prvi += m.getRezultat().prvi;
                r.drugi += m.getRezultat().drugi;
            }

            return r;
        }

        public bool isIgraGotova()
        {
            var r = getRezultat();
            if (r.treci == 0)
            {
                return r.prvi == r.drugi ? false : r.prvi > doKolikoSeIgra || r.drugi > doKolikoSeIgra;
            }
            else
            {
                if (r.prvi == r.drugi || r.prvi == r.treci || r.drugi == r.treci)
                {
                    return false;
                }

                if (r.prvi > doKolikoSeIgra || r.drugi > doKolikoSeIgra || r.treci > doKolikoSeIgra)
                {
                    return true;
                }

            }

            return false;
        }

    }
}

File: Kotlin/src/main/kotlin/model/Partija.kt
package model

/**
 * @author Matija Skrletović
 **/

abstract class Partija : Entitet() {

    abstract val doKolikoSeIgra: Int
    abstract val lokacija: Lokacija
    abstract val unosi: Igrac
    abstract val mjesanja: List<Mjesanje>
    abstract val igraci: MutableList<Igrac?>

    open fun getRezultat(): Rezultat {
        return Rezultat(
            mjesanja.stream().mapToInt { it.getRezultat()!!.prvi }.sum(),
            mjesanja.stream().mapToInt { it.getRezultat()!!.drugi }.sum()
        )
    }

    fun isIgraGotova(): Boolean {
        val rezultat = getRezultat()
        if (rezultat.isPocetak()) {
            return false
        }
        if (rezultat.treci == 0) {
            return if (rezultat.prvi == rezultat.drugi) false else rezultat.prvi > doKolikoSeIgra || rezultat.drugi > doKolikoSeIgra
        } else {
            if (rezultat.prvi == rezultat.drugi || rezultat.prvi == rezultat.treci || rezultat.drugi == rezultat.treci) {
                return false
            }
            if (rezultat.prvi > doKolikoSeIgra || rezultat.drugi > doKolikoSeIgra || rezultat.treci > doKolikoSeIgra) {
                return true
            }
        }
        return false
    }
}